src/MSIdentityScaffolding/Microsoft.DotNet.MSIdentity/Tool/MsAADTool.cs

```diff
@@ -67,42 +67,58 @@ internal async Task<string> PrintApplicationsList()
             var graphObjects = await GraphServiceClient.Me.OwnedObjects
                             .Request()
                             .GetAsync();
-            IList<Application> applicationList = new List<Application>();
-            if (graphObjects != null && graphObjects.Any())
+
+            if (graphObjects != null)
             {
-                foreach (var graphObj in graphObjects)
+                var graphObjectsList = graphObjects.ToList();
+
+                var nextPage = graphObjects.NextPageRequest;
+                while (nextPage != null)
                 {
-                    if (graphObj is Application app)
-                    {
-                        {
-                            applicationList.Add(app);
-                        }
-                    }
+                    var additionalGraphObjects = await nextPage.GetAsync();
```

Comment:
Error handling: Right now, if page 1 succeeds and page 2 fails, this throws and there are no results.
Generally, the best way to handle it is to return the partial results AND the fact that there was an error.  For commandline, you can use stderr for the body of the error (the thrown exception, I think) and stdout for the (partial) results.  For JSON, you might need to introduce a top level property for 'is partial results' and/or 'exception'.
For this PR, consider returning partial results (catch/stderr the exception/return what you have); I don't think Zach is planning on you changing the interface in this PR, but let's adjust for the next one that has an expected interface change.

Reply:
added try catch blocks with console messaging.